splinepy/bspline.py

```diff
@@ -4,8 +4,82 @@
 
 
 class BSplineBase(spline.Spline):
-    """BSpline base. Contains extra operations that's only available for
+    r"""BSpline base. Contains extra operations that are only available for
     bspline families.
+
+    All B-Spline-like splines feature knot vectors for each parametric
+    dimension. A knot vector can be considered as a set of non-decreasing
+    parametric coordinates. If coordinates in the parametric dimension are
+    denoted by :math:`u\in\mathbb{R}^{N_{param}}`, then the knot vector
+    associated with this dimension contains values
+
+    .. math::
+            \Theta_u = \{u_1,u_2,...,u_{l+p+1}\}
+
+    such that
+
+    .. math::
+            u_1 \leq u_2 \leq ... \leq u_{l+p+1}
+
+    Here, :math:`l` denotes the number of basis functions/control points along
+    this parametric dimension and :math:`p` the polynomial order of these
+    basis functions.
```

Comment:
I know it's a bit of an overkill, but can you add the relation of the expected number of control points, knots and the degree respectively